### PR TITLE
Invoke SonarQube via GitHub action

### PR DESCRIPTION
**Pull request by kaufmo**
## Acceptance criteria

-   [x] I have verified if my change requires [an example](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#example)
-   [x] I have verified if my change requires [a changeset](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#changeset)
-   [x] I have verified if my change requires [screenshots/screencasts](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md#screenshotsscreencasts)

### CODE REVIEW

**Review by manuelblum — APPROVED** [review]

**Review by manuelblum — DISMISSED** [review]

File: .github/workflows/sonar-qube.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+            - name: Use Node.js 22.x
+              uses: actions/setup-node@v4
+              with:
+                  node-version: 22
+                  registry-url: "https://registry.npmjs.org"
+                  cache: "pnpm"
+
+            - run: pnpm install --frozen-lockfile
+
```
Comment (johnnyomair) [inline .github/workflows/sonar-qube.yml:33]: Is this necessary? Shouldn't Sonar work without dependencies?

https://github.com/SonarSource/sonarcloud-github-action?tab=readme-ov-file#usage

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# This is the name and version displayed in the SonarCloud UI.
+#sonar.projectName=comet
+#sonar.projectVersion=1.0
+
+
+# Path is relative to the sonar-project.properties file. Replace "\" by "/" on Windows.
+#sonar.sources=.
+
+# Encoding of the source code. Default is default system encoding
+#sonar.sourceEncoding=UTF-8
```
Comment (johnnyomair) [inline sonar-project.properties:13]:
```suggestion
```

I'd remove these comments

File: .github/workflows/sonar-qube.yml (hunk 1 above)
Comment (kaufmo) [reply .github/workflows/sonar-qube.yml:33]: that is needed for the tsconfig files there are imports from packages inside the extends

File: sonar-project.properties (hunk 2 above)
Comment (kaufmo) [reply sonar-project.properties:13]: I've added this because sonar recommends this

**Review by johnnyomair — APPROVED** [review]

**Review by manuelblum — APPROVED** [review]

**Pull request merged**